magit-gh-comments.el

```diff
@@ -187,25 +305,155 @@ Github-style position."
+         (github-pos (magit-gh--diff-pos/magit->gh (magit-current-file) magit-pos diff-body))
+         (commit-sha (cdr (magit-split-range (magit-diff--dwim)))))
+    (magit-gh--post-pr-comment (magit-gh--get-current-pr)
+                               (magit-current-file) ;; FIXME: Add current-file to github-pos, get rid of this arg
                                commit-sha
                                github-pos
                                comment-text)))
 
+;; TODO: Make nice customizable faces
+(defgroup magit-gh-comments nil
+  "Add comments to Github Pull Requests from magit"
+  :group 'code)
+
+(defun magit-gh--format-comment-body (s &optional fill-width)
+  "Format the string S as the body of a Github comment.
+
+Fill the paragraph up to FILL-WIDTH characters."
+  (with-temp-buffer
+    (insert s)
+    (let ((fill-column (or fill-width 80)))
+      (fill-region (point-min) (point-max)))
+    (goto-char (point-min))
+    (while (re-search-forward "^" nil t)
+      (replace-match ">>> "))
+    (buffer-substring-no-properties (point-min) (point-max))))
+
+(defun magit-gh--propertize-overlay-text (s)
+  "Propertize the string S as a comment in a magit-diff buffer."
+  (propertize (concat (magit-gh--format-comment-body s) "\n")
+              'face 'magit-blame-highlight ;; TODO: Change this)
+              ))
+
+(defun magit-gh--display-comments (comments gh-diff-body)
+  "Display comments in the magit-diff buffer.
+
+Given a sequence of COMMENTS and Github diff GH-DIFF-BODY from
+which they came, add them to current magit-diff buffer."
+  (save-excursion
+    (dolist (comment comments)
+      (let ((magit-pos (magit-gh--diff-pos/gh->magit (alist-get :path comment)
+                                                     (alist-get :position comment)
+                                                     gh-diff-body))
+            (ov))
+        (magit-gh--visit-diff-pos (alist-get :path comment) magit-pos)
+        (setq ov (make-overlay (point-at-bol) (1+ (point-at-eol))))
+        (overlay-put ov 'magit-gh-comment comment)
+        (overlay-put ov 'face 'underline)
+        (overlay-put ov 'after-string (magit-gh--propertize-overlay-text
+                                       (format "%s - @%s"
+                                               (alist-get :body comment)
+                                               (alist-get :author comment))))))))
+
+(defun magit-gh--delete-comment-overlays ()
+  "Delete all comment overlays in the current buffer."
+  (interactive)
+  (let ((overlays (filter (lambda (ov) (overlay-get ov 'magit-gh-comment))
+                          (overlays-in (point-min) (point-max)))))
+    (dolist (overlay overlays)
+      (delete-overlay overlay))))
+
+;;;###autoload
+;; TODO: Assert that the current magit-gh-pull revision is up to date
+;; with HEAD of the PR branch
+(defun magit-gh-refresh-comments ()
+  "Refresh comments for the current PR."
```

Comment:
Where does this get called? It seems likes sometimes comments are left dangling when a PR diff window is re-opened.